ssz-rs/src/list.rs

```diff
@@ -188,6 +188,18 @@ where
     T: SimpleSerialize,
 {
     fn deserialize(encoding: &[u8]) -> Result<Self, DeserializeError> {
+        if !T::is_variable_size() {
```

Comment:
note, this check is not entirely redundant but if we want to remove the duplicity for `Vector` and array types, then we can do something like this